index.html

```diff
@@ -2989,6 +3015,88 @@ <h3 id="applying">
           </li>
         </ol>
       </section>
+      <section>
+        <h3>
+          <code>type</code> member
+        </h3>
+        <p>
+          The <dfn data-lt="serviceworker-type"><code>type</code> member</dfn> of a
+          <a>serviceworker object</a> is the service worker's
+          <a href="https://html.spec.whatwg.org/multipage/workers.html#workertype">worker type</a>.
+        </p>
+        <p>
+          The <dfn>steps for processing the <code>type</code> member of a
+          service worker</dfn> are given by the following algorithm. The algorithm takes
+          a <a>serviceworker object</a> <var>registration</var>, and a <a>URL</a> <var>manifest
+          URL</var>, which is the <a>URL</a> from which the
+          <var>manifest</var> was fetched. This algorithm will return a
+          string.
+        </p>
+        <ol>
+          <li>Let <var>value</var> be the result of calling the
+          <a>[[\GetOwnProperty]]</a> internal method of <var>registration</var>
+          passing " <code>type</code>" as the argument.
+          </li>
+          <li>Let <var>type</var> be <a>Type</a>(<var>value</var>).
+          </li>
+          <li>If <var>type</var> is not "string", then:
+            <ol>
+              <li>If <var>type</var> is not "<code>undefined</code>", issue a
+              developer warning that the type is unsupported.
+              </li>
+              <li>Return <code>"classic"</code>.
+              </li>
+            </ol>
+          </li>
+          <li>If <a>Trim</a>(value) is the empty string, then:
+            <ol>
+              <li>Issue a developer warning that the type is unsupported.
+              </li>
+              <li>Return <code>undefined</code>.
+              </li>
+            </ol>
+          <li>Return <var>type</var>.
```

Comment:
Ah.. I got it. It has to do with the prefix ! for an ECMAScript abstract operation. So, the _value_ here is the actual value for this member already. But then it still needs to return _value_ instead of _type_. Referring to other cases, it should return Trim(value) here I suppose.